chia/_tests/core/full_node/test_address_manager.py

```diff
@@ -542,36 +570,25 @@ async def test_serialization(self, tmp_path: Path):
         if peers_dat_filename.exists():
             peers_dat_filename.unlink()
         # Write out the serialized peer data
-        await AddressManagerStore.serialize(addrman, peers_dat_filename)
+        await AddressManagerStore.serialize_bytes(addrman, peers_dat_filename)
         # Read in the serialized peer data
         addrman2 = await AddressManagerStore.create_address_manager(peers_dat_filename)
-
-        retrieved_peers = []
-        for _ in range(50):
-            peer = await addrman2.select_peer()
-            if peer not in retrieved_peers:
-                retrieved_peers.append(peer)
-            if len(retrieved_peers) == 3:
-                break
-        assert len(retrieved_peers) == 3
         wanted_peers = [
             ExtendedPeerInfo(t_peer1, source),
             ExtendedPeerInfo(t_peer2, source),
             ExtendedPeerInfo(t_peer3, source),
         ]
-        recovered = 0
-        for target_peer in wanted_peers:
-            for current_peer in retrieved_peers:
-                if (
-                    current_peer is not None
-                    and current_peer.peer_info == target_peer.peer_info
-                    and current_peer.src == target_peer.src
-                    and current_peer.timestamp == target_peer.timestamp
-                ):
-                    recovered += 1
-        assert recovered == 3
+        assert await self.check_retrieved_peers(wanted_peers, addrman2)
         peers_dat_filename.unlink()
 
+    @pytest.mark.anyio
+    async def test_load_missing_file(self, tmp_path: Path):
+        peers_dat_filename = tmp_path / "peers.dat"
+        if peers_dat_filename.exists():
+            peers_dat_filename.unlink()
```

Comment:
as is there remains a race.  delete and ignore a not found error.

```suggestion
        with contextlib.suppress(FileNotFoundError):
            peers_dat_filename.unlink()
```

Reply:
I think this is a good suggestion. it also addresses a time-of-check-time-of-use issue